pkg/controller/flinkapplication/flink_state_machine.go

```diff
@@ -1201,7 +1210,6 @@ func (s *FlinkStateMachine) teardownApplicationVersion(ctx context.Context, appl
 		s.flinkController.LogEvent(ctx, application, corev1.EventTypeWarning, "TeardownFailed",
 			fmt.Sprintf("Failed to force-cancel application version %v and hash %s; will attempt to tear down cluster immediately: %s",
 				versionToTeardown, versionHashToTeardown, err))
-		return s.deployFailed(application)
```

Comment:
It should return something or exist from the method.

Reply:
No, I didn't make a mistake here. I did it on purpose. I don't want to interrupt a tearing down process of cluster due to the fact that there is an issue with force-canceling of job on that cluster. It is an unnecessarily operation, roughly.